internal/services/appservice/function_app_function_resource_test.go

```diff
@@ -0,0 +1,319 @@
+  name            = "testAcc-FnAppFn-%[2]d"
+  function_app_id = azurerm_linux_function_app.test.id
+  language        = "Python"
+  test_data = jsonencode({
+    "name" = "Azure"
+  })
+  config_json = jsonencode({
+    "bindings" = [
+      {
+        "authLevel" = "function"
+        "direction" = "in"
+        "methods" = [
+          "get",
+          "post",
+        ]
+        "name" = "req"
+        "type" = "httpTrigger"
+      },
+      {
+        "direction" = "out"
+        "name"      = "$return"
+        "type"      = "http"
+      },
+    ]
+  })
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r FunctionAppFunctionResource) basicUpdate(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_function_app_function" "test" {
+  name            = "testAcc-FnAppFn-%[2]d"
+  function_app_id = azurerm_linux_function_app.test.id
+  language        = "Python"
+  test_data = jsonencode({
+    "name" = "AzureRM"
+  })
+  config_json = jsonencode({
+    "bindings" = [
+      {
+        "authLevel" = "function"
+        "direction" = "in"
+        "methods" = [
+          "get",
+          "post",
+          "put",
+        ]
+        "name" = "req"
+        "type" = "httpTrigger"
+      },
+      {
+        "direction" = "out"
+        "name"      = "$return"
+        "type"      = "http"
+      },
+    ]
+  })
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r FunctionAppFunctionResource) withLocalFiles(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_function_app_function" "test" {
+  name            = "testAcc-FnAppFn-%[2]d"
+  function_app_id = azurerm_windows_function_app.test.id
+  language        = "CSharp"
+  file {
+    name    = "run.csx"
+    content = file("testdata/run.csx")
+  }
+  test_data = jsonencode({
+    "name" = "Azure"
+  })
+  config_json = jsonencode({
+    "bindings" = [
+      {
+        "authLevel" = "function"
+        "direction" = "in"
+        "methods" = [
+          "get",
+          "post",
+        ]
+        "name" = "req"
+        "type" = "httpTrigger"
+      },
+      {
+        "direction" = "out"
+        "name"      = "$return"
+        "type"      = "http"
+      },
+    ]
+  })
+}
+`, r.templateWindows(data), data.RandomInteger)
+}
+
+func (r FunctionAppFunctionResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_function_app_function" "import" {
+  name            = azurerm_function_app_function.test.name
+  function_app_id = azurerm_function_app_function.test.function_app_id
+  language        = azurerm_function_app_function.test.language
+  test_data       = azurerm_function_app_function.test.test_data
+  config_json     = azurerm_function_app_function.test.config_json
+}
+
+`, r.basic(data))
+}
+
+func (r FunctionAppFunctionResource) template(data acceptance.TestData) string {
```

Comment:
pedantic, but as Windows is labelled below we should do the same here:

```suggestion
func (r FunctionAppFunctionResource) templateLinux(data acceptance.TestData) string {
```